versioned_docs/version-v2/framework-integration/react.md

```diff
@@ -12,199 +12,483 @@ contributors:
+to re-export the Stencil loader's `defineCustomElements()` method. In your React library's entry file (`packages/react-library/lib/index.ts`),
+add the follow:
 
-```bash npm2yarn
-npm run build
+```diff
+export * from "./components/stencil-generated";
++ export { defineCustomElements } from "stencil-library/loader";
 ```
 
-Now you can create a symlink for your Stencil component library. From within your **Stencil component library**, run
+### Link Your Packages (Optional)
+
+:::note
+If you are using a monorepo tool (Lerna, Nx, etc.), skip this section.
+:::
+
+Before you can successfully build a local version of your React component library, you will need to link the Stencil package to the React package.
+
+From your Stencil project's directory, run the following command:
 
 ```bash npm2yarn
+# Link the working directory
 npm link
 ```
 
-With the symlink created, we next need to specify which packages will be consuming them. Your React component library will need to consume your Stencil component library. In the directory of your **React component library** run
+From your React component library's directory, run the following command:
 
 ```bash npm2yarn
-npm link {Stencil library name}
+# Link the package name
+npm link name-of-your-stencil-package
 ```
 
-And with that, your component libraries are linked together. Now, you can make changes in your Stencil component library and run `npm run build` to propagate them through to the React component library without having to relink.
+The name of your Stencil package should match the `name` property from the Stencil component library's `package.json`.
+
+Your component libraries are now linked together. You can make changes in the Stencil component library and run `npm run build` to propagate the
+changes to the React component library.
 
 :::note
-As an alternative to `npm link` , you can also run `npm install` with a relative path to your Stencil component library. This strategy, however, will modify your `package.json` so it is important to make sure you do not commit those changes.
+As an alternative to `npm link` , you can also run `npm install` with a relative path to your Stencil component library. This strategy, however, will
+modify your `package.json` so it is important to make sure you do not commit those changes.
 :::
 
-## Usage
+## Consumer Usage
 
-If you are developing and testing your React component library locally, you'll have to use `npm link` again to make your React component library available in your React application. If your components are published to npm, you can skip this step.
+### Creating a Consumer React App
 
-To link your React component library, navigate to your **React component library** and run
+:::note
+If you already have a React app, skip this section.
+:::
 
-```bash npm2yarn
-npm run build
-npm link
+From the `packages/` directory, run the following command to create a starter React app:
+
+```bash
```

Comment:
First off, I decided to swap this off of `create-react-app` to a Vite React app instead in [4d88bc3](https://github.com/ionic-team/stencil-site/pull/964/commits/4d88bc3a9b31d628edce650bdb62cbb66793c77d). Did this for two reasons:
1. The command for generating w/ `create-react-app` has some issues in Yarn environments
2. `create-react-app` is kinda dead, so it's easy enough for us to move away from it

Even so, we can't use `npm2yarn` here because that plugin can't correctly convert certain syntaxes (like `npm create`/`npm init`) to yarn syntax. It tries its best but then spits out a comment in the generated yarn block along the lines of "couldn't auto-convert command". I believe it's related to [this issue](https://github.com/facebook/docusaurus/issues/5861) in the docusaurus repo. I marked a TODO comment in the docs tho for us to look into using npm2yarn if/when that is ever resolved